packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,553 @@
+      preferredContentSize: preferredContentSize,
+      delegate: delegate ?? RegularWindowControllerDelegate(),
+    );
+    if (title != null) {
+      controller.setTitle(title);
+    }
+    return controller;
+  }
+
+  /// Creates an empty [RegularWindowController].
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  // TODO(mattkae): Replace @internal with @visibleForTesting when this API is non-experimental
+  @internal
+  @protected
+  RegularWindowController.empty();
+
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  @override
+  WindowType get type => WindowType.regular;
+
+  /// Request change for the window content size.
+  ///
+  /// [contentSize] describes the new requested window size. The properties
+  /// of this object are applied independently of each other. For example,
+  /// setting [WindowSizing.preferredSize] does not affect the [WindowSizing.preferredConstraints]
+  /// set previously.
+  ///
+  /// The platform is free to ignore the request.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void updateContentSize(WindowSizing sizing);
+
+  /// Request change for the window title.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setTitle(String title);
+
+  /// Requests that the window be displayed in its current size and position.
+  ///
+  /// If the window is minimized, the window returns to the size  and position
+  /// that it had before that state was applied. The window will also be
+  /// brought to the top of the window stack.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void activate();
+
+  /// Requests the window to be maximized.
+  ///
+  /// This has no effect if the window is currently full screen or minimized,
+  /// but might affect the window size upon restoring it from minimized or
+  /// full screen state.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setMaximized(bool maximized);
+
+  /// Returns whether window is currently maximized.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  bool isMaximized();
+
+  /// Requests window to be minimized.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setMinimized(bool minimized);
+
+  /// Returns whether window is currently minimized.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  bool isMinimized();
+
+  /// Request change for the window to enter or exit fullscreen state.
+  ///
+  /// If [fullscreen] is set to true, the platform will attempt to change
+  /// the state of the window to fullscreen. If false, the window will
+  /// return to a previous non-hidden state. Both cases might not be
+  /// honored by the platform.
+  ///
+  /// The [display] specifies an optional [Display] on which the window
+  /// would like to be fullscreened. This might not be honored by the
+  /// platform.
```

Comment:
I imagine `display` is ignored if `fullscreen` if `false`? Consider adding:


```suggestion
  /// platform. The [display] argument is ignored if [fullscreen] is `false`.
```